source/text/Format.cpp

```diff
@@ -389,3 +417,99 @@ vector<string> Format::Split(const string &str, const string &separator)
 	}
 	return result;
 }
+
+
+
+string Format::ExpandConditions(const string &source, ConditionGetter getter)
+{
+	string result;
+	result.reserve(source.size());
+
+	size_t formatStart = string::npos;
+	size_t formatSize = string::npos;
+	size_t conditionStart = string::npos;
+	size_t conditionSize = string::npos;
+
+	// Hand-coded regular grammar parser for:
+	//	&[format@condition]
+	//	&[condition]
+	// Using these states:
+	//	state = _ ----- outside of all &[] regions
+	//	state = & ----- just read & and hoping to see a [
+	//	state = [ ----- read &[ but haven't seen @ or ] yet
+	//	state = N ----- inside a nested [] of depth `depth`
+	//	state = @ ----- read &[...@ but haven't seen ] yet. Have format start & size.
+	// Anything inside a &[...] is sent to AppendCondition
+	char state = '_';
+	// Depth of nested [] within the &[...]
+	int depth = 0;
+	// State before entering the nested []
+	char oldState = '[';
+	// "start" is the beginning of the text that has not yet been sent to result.
+	size_t start = 0;
+	for(size_t look = 0; look < source.size(); ++look)
+	{
+		char next = source[look];
+		// This would be faster with a nested select, but that would be
+		// harder to read, and I don't expect this to be performance-critical.
+		if(state == '_' && next == '&')
+			state = '&';
+		else if(state == '_' || (state == '&' && next != '['))
```

Comment:
Make these repeated char literals static const at the top of this method.
For example: `static const char PREFIX = '&';`
And refer to those instead of repeating the literals.

Reply:
That would be less readable. With a char, you know exactly what character is being searched for.

Reply:
I have implemented it, but I think the result is far less readable.